touchforms/backend/persistence.py

```diff
@@ -24,18 +25,18 @@ def restore(sess_id, factory, override_state=None):
         state.update(override_state)
     return factory(**state)
 
-# TODO integrate with real caching framework (django ideally)
-def cache_set(key, value, timeout):
-    with open(cache_path(key), 'w') as f:
-        f.write(json.dumps(value).encode('utf8'))
 
-# TODO integrate with real caching framework (django ideally)
+def cache_set(key, value):
+    if key is None:
+        raise KeyError
+    postgres_insert(key, value)
+
+
 def cache_get(key):
-    try:
-        with open(cache_path(key)) as f:
-            return json.loads(f.read().decode('utf8'))
-    except IOError:
+    if key is None:
         raise KeyError
+    try:
+        return postgres_lookup(key)
     except JSONDecodeError:
         raise EmptyCacheFileException(_(
             u"Unfortunately an error has occurred on the server and your form cannot be saved. "
```

Comment:
this was the message on the old cache_get method just copied over, happy to
change

On Wed, Feb 25, 2015 at 6:02 PM, Cory Zue <notifications@github.com> wrote:

> In touchforms/backend/persistence.py
> <https://github.com/dimagi/touchforms/pull/82#discussion_r25389049>:
>
> >          raise KeyError
> > +    try:
> > +        return postgres_lookup(key)
> >      except JSONDecodeError:
> >          raise EmptyCacheFileException(_(
> >              u"Unfortunately an error has occurred on the server and your form cannot be saved. "
>
> this message also seems wrong for pulling something out. shouldn't it be
> "retreived"?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/touchforms/pull/82/files#r25389049>.
>